src/app/pipes/filter-boards.pipe.ts

```diff
@@ -21,11 +24,15 @@ export class FilterBoardsPipe implements PipeTransform {
       );
     }
     if (param === 'task') {
-      return copy.filter((task) =>
-        filter
-          ? task.name.toLowerCase().includes(filterStr.toLowerCase())
-          : task
-      );
+      return copy.filter((board: any) => {
+        if (filter) {
+          if (filterThroughTasks(board.tasks, filterStr).length > 0) {
+            return board;
+          } else {
+            return;
+          }
+        }
```

Comment:
for filter function you need to return something
in case of !filter - add return